src/api/app/services/action_sourcediff_parser.rb

```diff
@@ -0,0 +1,74 @@
+class ActionSourcediffParser
+  attr_accessor :action
+
+  def initialize(action_sourcediff:)
+    @action_sourcediff = action_sourcediff
+  end
+
+  def call
+    @action_sourcediff&.each do |sourcediff|
+      sourcediff['filenames']&.each do |filename|
+        content = sourcediff['files'][filename].dig('diff', '_content')
+
+        sourcediff['files'][filename]['diff']['parsed_lines'] = parse_file_diffs(content)
+      end
+    end
+  end
+
+  private
+
+  def parse_file_diffs(content)
+    @original_index = 0
+    @changed_index = 0
+    @last_original_index = 0
+    @last_changed_index = 0
+    lines = []
+    content&.each_line&.with_index do |line, line_index|
```

Comment:
```suggestion
    content&.each_line.with_index do |line, line_index|
```
I believe this ampersand is not needed as applying the change above, you no longer receive `nil` in content.

Reply:
True. Done.